api/scpca_portal/models/dataset.py

```diff
@@ -67,6 +78,105 @@ class Meta:
     def __str__(self):
         return f"Dataset {self.id}"
 
+    @classmethod
+    def get_ccdl_dataset(cls, ccdl_name):
+        return cls(is_ccdl=True, ccdl_name=ccdl_name)
+
+    def populate_ccdl_data(self, projects: Iterable[Project]):
```

Comment:
```suggestion
    def update_ccdl_data(self):
```

Reply:
And then update the method to use the ccdl attributes and return the dict instead of assigning it.